File: djisktra&co/Java/src/graph/Node.java
/**
 * Proiectarea Algoritmilor, 2014
 * Lab 8: Drumuri minime
 * 
 * @author 	Radu Iacob
 * @email	radu.iacob23@gmail.com
 */

package graph;

public class Node {

	private String _city;
	
	private int _id;
	private boolean _visited = false;
	private Node parent = null;
	
	public Node(String city, int id)
	{
		_city = city;
		_id   = id;
	}
	
	public Node(int id)
	{
		_city = "";
		_id = id;
	}
	
	public void setCity( String city ){
		_city = city;
	}
	
	public String getCity(){
		return _city;
	}
	
	public int getId(){
		return _id;
	}

	public boolean isVisited(){
		return _visited;
	}
	
	public void visit(){
		_visited = true;
	}

	public void reset(){
		_visited = false;
	}
	
	public Node getParent() {
		return parent;
	}

	public void setParent(Node parent) {
		this.parent = parent;
	}
	
	public String toString()
	{
		String res = "Node : ";
		if( _city.length() != 0 ) res += _city + " , ";
		return res += (_id);
	}

}

File: djisktra&co/C++/Node.cpp
/**
 * Proiectarea Algoritmilor, 2014
 * Lab 8: Drumuri minime
 * 
 * @author 	Radu Iacob
 * @email	radu.iacob23@gmail.com
 */
 
#include "Node.h"

Node::Node(uint id)
	: _city(""), _id(id), visited(false), parent(NULL)
{
}

Node::Node(std::string city, uint id)
	: _city(city), _id( id ), visited(false), parent(NULL)
{
}

void Node::set_parent( Node* parent )
{
	this->parent = parent;
}

Node* Node::get_parent() const
{
	return parent;
}


const uint Node::get_id() const
{
	return _id;
}

const std::string& Node::get_city() const
{
	return _city;
}

void Node::visit()
{
	visited = true;
}

bool Node::is_visited() const
{
	return visited;
}

void Node::reset()
{
	visited = false;
}

std::istream& operator>> (std::istream& in, Node& node)
{
	in >> node._city;
	return in;
}

std::ostream& operator<< (std::ostream& out, Node& node )
{
	out << "Node : " << node.get_city();
	return out;
}

std::ostream& operator<< (std::ostream& out, Node* node )
{
	out << *node;
	return out;
}
